### PR TITLE
fix version wrangling in docs

### PR DESCRIPTION
**Pull request by pmeier**
We have some logic in our docs that handles the version wrangling for released versions:

https://github.com/pytorch/vision/blob/abc40ef241e309bff598222b1accff8d0eb93e8d/docs/source/conf.py#L96-L105

Two issues here:

1. To enable the wrangling, we need to set the `VERSION` environment value. That is currently not the case.
2. The logic is broken. Since we wrangle after we have already prefixed the version with `main(`, this is retained for the final version.

This PR fixes both issues. I'll send a follow-up PR to the `gh-pages` branch to remediate this for the past two versions.

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -43,6 +43,14 @@ jobs:
         pip install --progress-bar=off -r requirements.txt
         echo '::endgroup::'
         
+        if [[ ${{ github.ref_type }} == tag || (${{ github.ref_type }} == branch && ${{ github.ref_name }} == release/*) ]]; then
+          echo '::group::Enable version wrangling'
+          # This environment variable just has to exist and must not be empty. The actual value is arbitrary.
+          # See docs/source/conf.py for details
+          export VERSION=1
```
Comment (NicolasHug) [inline .github/workflows/docs.yml:50]: should we call this `SANITIZE_VERSION_IN_DOCS` or something like that?

**Review by NicolasHug — APPROVED** [review]
Thanks a lot for fixing this Philip, this has been a bit of a pain to manually clean that after every release (and we often forget). This is mostly a stamp but if there's any issue they'll be easy to address

File: .github/workflows/docs.yml (hunk 1 above)
Comment (pmeier) [reply .github/workflows/docs.yml:50]: Happy to do so. Let me see if anything else relies on this specific name of the env var.

Comment (pmeier) [reply .github/workflows/docs.yml]: I've `grep`'ed through the repository and it seems this env var is not used anywhere else. Meaning, we are good to go. I've used `TORCHVISION_SANITIZE_VERSION_STR_IN_DOCS`

**Pull request merged**